File: for-local-test/383.ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int [] count = new int[26];
        for(char c : magazine.toCharArray()){
            count[c - 'a']++;
        }
        for(char c : ransomNote.toCharArray()){
            if (count[c - 'a'] == 0) return false;
            count[c - 'a']--;
        }
        return true;
    }
    public static void main(String[] args) {
        Solution solution = new Solution();
        String ransomNote = "a";
        String magazine = "ba";
        boolean result = solution.canConstruct(ransomNote, magazine);
        System.out.println(result); // Output: true
    }
}

File: for-local-test/python/383.ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        count = [0] * 26
        for c in magazine:
            count[ord(c) - ord('a')] += 1
        for c in ransomNote:
            if count[ord(c) - ord('a')] == 0:
                return False
            count[ord(c) - ord('a')] -= 1
        return True
    
if __name__ == "__main__":
    solution = Solution()
    ransomNote = "a"
    magazine = "ba"
    result = solution.canConstruct(ransomNote, magazine)
    print(result)  # Output: True